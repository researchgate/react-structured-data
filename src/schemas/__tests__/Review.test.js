import React from "react";
import renderer from "react-test-renderer";
import {
  AdministrativeArea,
  JSONLD,
  Person,
  Product,
  Rating,
  Review,
  ReviewCollection
} from "../../index";

describe("Review Page", () => {
  it("renders reviews in graph", () => {
    expect(
      renderer.create(
        <JSONLD>
          <ReviewCollection type="@graph">
            <Review
              name="It's awesome"
              reviewBody="This is Great! My family loves it"
              datePublished="11/22/1963"
            >
              <Person type="author" name="Jerry" />
              <AdministrativeArea type="locationCreated" name="Chicago, IL" />
              <Rating type="reviewRating" ratingValue={5} />
              <Product type="itemReviewed" name="Product Name" id="product-x" />
            </Review>
            <Review
              name="Very cool"
              reviewBody="I like this a lot. Very cool product"
              datePublished="11/22/1963"
            >
              <Person type="author" name="Cool Carl" />
              <AdministrativeArea type="locationCreated" name="Chicago, IL" />
              <Rating type="reviewRating" ratingValue={4} />
              <Product type="itemReviewed" name="Product Name" id="product-x" />
            </Review>
          </ReviewCollection>
        </JSONLD>
      )
    ).toMatchSnapshot();
  });
});
