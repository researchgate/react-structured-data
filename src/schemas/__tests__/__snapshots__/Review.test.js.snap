// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Page renders reviews in graph 1`] = `
<script
  type="application/ld+json"
>
  {"@context":"https://schema.org/","@graph":[{"@type":"Review","name":"It's awesome","reviewBody":"This is Great! My family loves it","datePublished":"11/22/1963","author":{"@type":"Person","name":"Jerry"},"locationCreated":{"@type":"AdministrativeArea","name":"Chicago, IL"},"reviewRating":{"@type":"Rating","ratingValue":5},"itemReviewed":{"@type":"Product","name":"Product Name","@id":"product-x"}},{"@type":"Review","name":"Very cool","reviewBody":"I like this a lot. Very cool product","datePublished":"11/22/1963","author":{"@type":"Person","name":"Cool Carl"},"locationCreated":{"@type":"AdministrativeArea","name":"Chicago, IL"},"reviewRating":{"@type":"Rating","ratingValue":4},"itemReviewed":{"@type":"Product","name":"Product Name","@id":"product-x"}}]}
</script>
`;
